util/system/path.go

```diff
@@ -13,16 +13,36 @@ import (
 // ':' character .
 const DefaultPathEnvUnix = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 
-// DefaultPathEnvWindows is windows style list of directories to search for
-// executables. Each directory is separated from the next by a colon
-// ';' character .
-const DefaultPathEnvWindows = "c:\\Windows\\System32;c:\\Windows"
+// DefaultPathEnvWindows is deliberately empty on Windows and the default path
+// must be set by the image. BuildKit has no context of what the default
+// path should be.
+//
+// Deprecated: Windows images must not have a default PATH set.
+const DefaultPathEnvWindows = ""
 
-func DefaultPathEnv(os string) string {
+// DefaultPathEnv returns the default value for the PATH environment-variable
+// if available for the given operating-system. The second return variable
+// indicates whether a default exists.
+//
+// On Linux/Unix, this returns [DefaultPathEnvUnix]. On Windows, no default
+// exists, and PATH must not be set as its image-dependent, and must be
+// configured through other means.
+//
+// More details about default PATH values for Windows images can be found
+// in the following GitHub discussions:
+//
+// - [moby/moby#13833]
+// - [moby/buildkit#3158]
+// - [containerd/containerd#9118]
+//
+// [moby/moby#13833]: https://github.com/moby/moby/pull/13833
+// [moby/buildkit#3158]: https://github.com/moby/buildkit/pull/3158
+// [containerd/containerd#9118]: https://github.com/containerd/containerd/pull/9118
+func DefaultPathEnv(os string) (string, bool) {
 	if os == "windows" {
-		return DefaultPathEnvWindows
+		return "", false
 	}
-	return DefaultPathEnvUnix
+	return DefaultPathEnvUnix, true
 }
```

Comment:
And updated this function to return a bool to indicate whether there is a default; also tried to capture some of this in the function's GoDoc, and links to relevant discussions.

Reply:
Looks like doing so broke tests; is `PATH` already set somehow? (but to a linux default path?)

<img width="1013" alt="Screenshot 2023-11-06 at 09 46 07" src="https://github.com/moby/buildkit/assets/1804568/93416f06-ddaa-4b47-af20-e683800c2a86">

Reply:
Ah, hm, right, so test is running a `busybox` image on Linux; https://github.com/moby/buildkit/blob/6a0cd7cc8644c4f2baa690f631325dc99ef5e128/client/build_test.go#L1370-L1371

And checks the output of `echo $PATH`, with the expectation that that is changed because of the `windows` argument 🤔; https://github.com/moby/buildkit/blob/6a0cd7cc8644c4f2baa690f631325dc99ef5e128/client/build_test.go#L1418-L1421

Reply:
Hmm, yeah. Seems like an inappropriate test to me, since we don't actually support `RUN` steps for Windows images on non-Windows AFAIK. In fact, why doesn't `c.NewContainer` or `ctr.Start` _fail_ here? It seems weird to me that this ever got this far, but maybe the "we don't support this" or "inconsistent request" failure-handling is at a higher level.

We will probably want Windows-side tests to ensure that we aren't breaking `PATH` (including covering `setx /M` and `setx` across a couple of users to ensure we aren't leaking `PATH` changes across users. Since we don't run the integration test suite on Windows yet, we can probably defer adding that test, and just drop this one.